pkg/mpc-examples/contracts/04-pause-unpause-controller/PauseUnpauseController.sol

```diff
@@ -0,0 +1,118 @@
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract PauseUnpauseController is Ownable {
+    uint256 private constant _END_SWAP_FEE_PERCENTAGE = 3e15;
+    uint256 private constant _START_SWAP_FEE_PERCENTAGE = 95e16;
+    uint256 private constant _SWAP_FEE_REDUCATION_DURATION = 3 days;
+
+    IVault private _vault;
+    bytes32 private _poolId;
+
+    constructor(IVault vault, address controllerOwner) {
+        // Get poolId from the factory.
+        bytes32 poolId = IManagedPool(ILastCreatedPoolFactory(msg.sender).getLastCreatedPool()).getPoolId();
+
+        // Verify that this is a real Vault and the pool is registered - this call will revert if not.
+        vault.getPool(poolId);
+
+        // store Vault and PoolId.
+        _poolId = poolId;
+        _vault = vault;
+
+        // Transfer ownership from factory to manager.
+        transferOwnership(controllerOwner);
+    }
+
+    /// === public Getters ===
+
+    function getPoolId() public view returns (bytes32) {
+        return _poolId;
+    }
+
+    function getVault() public view returns (IVault) {
+        return _vault;
+    }
+
+    function getSwapEnabled() public view returns (bool) {
+        return _getPool().getSwapEnabled();
+    }
+
+    function isPoolPaused() public view returns (bool) {
+        return !_getPool().getSwapEnabled();
+    }
+
+    /// === Setters ===
+
+    /**
+     * @notice Disables swapping.
+     */
+    function pausePool() external onlyOwner returns (bool) {
+        require(getSwapEnabled(), "swapping with pool is already paused");
+        _getPool().setSwapEnabled(false);
+        // pool is confirmed paused
+        return true;
+    }
+
+    /**
+     * @notice Unpauses the pool in a safe or unsafe manner.
+     * @dev A safe unpause is desirable as the market likely had price movements
+     * which have not been reflected in a paused pool. In order to not leak too much
+     * arbitrage losses, the controller adjusts swap fees of the managed pool to
+     * _START_SWAP_FEE_PERCENTAGE instantly and let's the market bring the pool back
+     * into balance via minimal viable arbitrage.
+     * @param shouldSafeUnpause Decision to safely unpause the pool.
+     */
+
+    /* solhint-disable not-rely-on-time */
+    function unpausePool(bool shouldSafeUnpause) external onlyOwner returns (bool) {
+        if (shouldSafeUnpause) {
+            _getPool().updateSwapFeeGradually(
+                block.timestamp,
+                block.timestamp + _SWAP_FEE_REDUCATION_DURATION,
+                _START_SWAP_FEE_PERCENTAGE,
+                _END_SWAP_FEE_PERCENTAGE
+            );
+            // Enabling swaps again after having updated the swap fee gradually is fine
+            // even if the the start time is at a future time.
+            _getPool().setSwapEnabled(true);
+            return true;
+        } else {
+            _getPool().setSwapEnabled(true);
+            return true;
+        }
```

Comment:
Instead of repeating the last two lines here, you could just move it outside the conditional block.
```
if (...) {
    ...
    _getPool().setSwapEnabled(true);
    return true;
} else {
    _getPool().setSwapEnabled(true);
    return true;
}
```
to be this:
```
if (...) {
    ...
}
_getPool().setSwapEnabled(true);
return true;
```